src/function/uvc.c

```diff
@@ -138,6 +139,131 @@ struct {
 		.export = usbg_set_config_node_int,		        \
 	}
 
+int h2b(unsigned char c)
+{
+	/* check if char is a number */
+	if ((c >= '0') && (c <= '9'))
+		return c - '0';
+
+	/* check if char is upper case */
+	if ((c >= 'A') && (c <= 'Z'))
+		return c - 'A' + 10;
+
+	/* check if char is lower case */
+	if ((c >= 'a') && (c <= 'z'))
+		return c - 'a' + 10;
+
+	return -EINVAL;
+}
+
+int usbg_write_guid(const char *path, const char *name,
+		      const char *file, const char *buf)
+{
+	int ret;
+	char guid[16];
```

Comment:
You are using 16 as the GUID length all over the place. Maybe it is worth adding a constant for that?

Reply:
Good point. I will add an define.

Reply:
Done